src/core/Akka.Remote/Transport/AkkaPduCodec.cs

```diff
@@ -376,13 +381,24 @@ public override ByteString ConstructDisassociate(DisassociateInfo reason)
             }
         }
 
+        private static ByteString _heartbeatPdu;
+
         /// <summary>
         /// TBD
         /// </summary>
         /// <returns>TBD</returns>
         public override ByteString ConstructHeartbeat()
         {
-            return ConstructControlMessagePdu(CommandType.Heartbeat);
+            /*
+             * Since there's never any ActorSystem-specific information coded directly
+             * into the heartbeat messages themselves (i.e. no handshake info,) there's no harm in caching in the
+             * same heartbeat byte buffer and re-using it.
+             */
```

Comment:
As explained in the comment above: `Heartbeat` messages don't contain any handshake information, therefore the same "heartbeat" message can be used across many different associations and even many different `ActorSystems` without having to be created from scratch each time. We cache this value in a static mutable field on the serializer now. We could probably make this immutable....